app/Cardano/Shell/Features/Logging.hs

```diff
@@ -68,44 +71,48 @@ createLoggingFeature cardanoEnvironment cardanoConfiguration = do
     -- the filesystem, so we give him the most flexible/powerful context, @IO@.
     loggingConfiguration    <-  LoggingParameters <$> (Config.setup $ ccLogConfig cardanoConfiguration)
 
+    (baseTrace, switchBoard) <- setupTrace_ (lpConfiguration loggingConfiguration) "cardano"
```

Comment:
You send the configuration to `loggingCardanoFeatureInit` and you setup the trace when starting the module? 
Then it will make more sense.
We are splitting the logic of a module into two distinct phases:
- initialization
- running

https://github.com/input-output-hk/cardano-shell/blob/develop/ARCHITECTURE.md